lua/astrocommunity/recipes/auto-session-restore/init.lua

```diff
@@ -0,0 +1,33 @@
+local read_from_stdin = false
+return {
+  {
+    "AstroNvim/astrocore",
+    ---@type AstroCoreOpts
+    opts = {
+      autocmds = {
+        -- disable alpha autostart
+        alpha_autostart = false,
+        restore_session = {
+          {
+            event = "StdinReadPost",
+            desc = "Check if we are reading from stdin",
+            nested = true,
+            callback = function() read_from_stdin = true end,
+          },
+          {
+            event = "VimEnter",
+            desc = "Restore previous directory session if neovim opened with no arguments",
+            nested = true, -- trigger other autocommands as buffers open
+            callback = function()
+              -- Only load the session if nvim was started with no args
+              if not read_from_stdin and vim.fn.argc(-1) == 0 then
+                -- try to load a directory session using the current working directory
+                require("resession").load(vim.fn.getcwd(), { dir = "dirsession", silence_errors = true })
+              end
+            end,
+          },
+        },
+      },
+    },
+  },
+}
```

Comment:
```suggestion
return {
  "AstroNvim/astrocore",
  ---@type AstroCoreOpts
  opts = {
    autocmds = {
      -- disable alpha autostart
      alpha_autostart = false,
      restore_session = {
        {
          event = { "VimEnter", "StdinReadPost" },
          desc = "Restore previous directory session if neovim opened with no arguments",
          once = true, -- delete itself after executing once
          nested = true, -- trigger other autocommands as buffers open
          callback = function(args)
            -- Only load the session if nvim was started with no args
            if args.event == "VimEnter" and vim.fn.argc(-1) == 0 then
              -- try to load a directory session using the current working directory
              require("resession").load(vim.fn.getcwd(), { dir = "dirsession", silence_errors = true })
            end
          end,
        },
      },
    },
  },
}
```

I think this approach may be a bit faster and also avoids the necessity for a separate variable and multiple autocommands. Rather than having a variable we have the autocommand trigger on both events. If the `StdinReadPost` happens then it simply deletes itself, if not then the `VimEnter` happens and it restores the session and then deletes itself.

Reply:
Let me know what you think

Reply:
Sorry, it does not work. It is not removed on `once=true`. I see it triggers twice - once for StdinReadPost and then for VimEnter. So the `once=true` does not protect it.

Reply:
Ok, I have a better way then. Standby

Reply:
Yea. So I think every event is registered separately, so `autocmd ++once` is evaluated for every event separately. I think it is the loop here https://github.com/neovim/neovim/blob/ba70404c558169b813f51d5a0bb71cd540555c93/src/nvim/api/auto cmd.c#L488 .